src/nvme/tree.c

```diff
@@ -1589,13 +1579,15 @@ static void _candidate_free(struct candidate_args *candidate)
 	freeifaddrs(candidate->iface_list); /* This is NULL-safe */
 }
 
+#define _cleanup_candidate_ __cleanup__(_candidate_free)
+
 nvme_ctrl_t __nvme_lookup_ctrl(nvme_subsystem_t s, const char *transport,
 			       const char *traddr, const char *host_traddr,
 			       const char *host_iface, const char *trsvcid,
 			       const char *subsysnqn, nvme_ctrl_t p)
 {
 	struct nvme_ctrl *c, *matching_c = NULL;
-	struct candidate_args candidate;
+	_cleanup_candidate_ struct candidate_args candidate;
```

Comment:
` = {}` ?

Reply:
There's no early `return`, so it's getting cleaned up at the end of the function just as in the existing code. Both the existing code and the new code are depending on `_candidate_init()` to initialize it, which is perhaps bad practice. So I don't think there's any behavior change, but I'm happy to add an explicit initializer if you'd prefer.

Reply:
Ah okay, so it's the same as in the other places where you left the initializer away. I don't have any preferences. Adding explicit initializer will obviously make the code a bit robuster when someone is factoring stuff around. Let's see how things are going. Adding them is pretty easy :)